server/src/main/java/org/elasticsearch/rest/action/search/RestSearchAction.java

```diff
@@ -208,6 +209,9 @@ public static void parseSearchRequest(
                 request.paramAsBoolean("ccs_minimize_roundtrips", searchRequest.isCcsMinimizeRoundtrips())
             );
         }
+        if (IndexSettings.isTimeSeriesModeEnabled() && request.paramAsBoolean("force_synthetic_source", false)) {
```

Comment:
Just to check: `IndexSettings.isTimeSeriesModeEnabled()` is the feature flag, yes?